infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/SshKeysProvisioner.java

```diff
@@ -84,27 +86,46 @@
 
   private static final String SSH_SECRET_TYPE = "opaque";
 
-  private static final Logger LOG = LoggerFactory.getLogger(VcsSshKeysProvisioner.class);
+  private static final Logger LOG = LoggerFactory.getLogger(SshKeysProvisioner.class);
 
   private final SshManager sshManager;
 
   @Inject
-  public VcsSshKeysProvisioner(SshManager sshManager) {
+  public SshKeysProvisioner(SshManager sshManager) {
     this.sshManager = sshManager;
   }
 
   @Override
   @Traced
   public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity identity)
       throws InfrastructureException {
-    TracingTags.WORKSPACE_ID.set(identity::getWorkspaceId);
+    String workspaceId = identity.getWorkspaceId();
+    TracingTags.WORKSPACE_ID.set(workspaceId);
 
+    List<SshPairImpl> vcsSshPairs = getVcsSshPairs(identity);
+    List<SshPairImpl> systemSshPairs = getSystemSshPairs(identity);
+
+    List<SshPairImpl> all = new ArrayList<>(vcsSshPairs);
+    all.addAll(systemSshPairs);
+
+    doProvisionSshKeys(all, k8sEnv, workspaceId);
+    doProvisionVcsSshConfig(vcsSshPairs, k8sEnv, workspaceId);
+  }
+
+  /**
+   * Return list of keys related to the VCS (Version Control Systems), Git, SVN and etc. Usually
+   * managed by user
+   *
+   * @param identity
+   * @return list of ssh pairs
+   */
+  private List<SshPairImpl> getVcsSshPairs(RuntimeIdentity identity) {
     List<SshPairImpl> sshPairs;
     try {
       sshPairs = sshManager.getPairs(identity.getOwnerId(), "vcs");
     } catch (ServerException e) {
       LOG.warn("Unable to get SSH Keys. Cause: {}", e.getMessage());
-      return;
+      return emptyList();
```

Comment:
throw InfrastructureException ?

Reply:
In generic existence SSH keys not required, if workspace service don't use any keys we can  start workspace even if SSH service is out of order.

Reply:
fixed: throw exception. ServerException means potential problems for workspace in totally so better indicate it earlier